### PR TITLE
CheckoutV2: Support processing channel and marketplace sub entity ID

### PR DESCRIPTION
**Pull request by AMHOL**
# Test Summary

## Unit
5034 tests, 74937 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## Rubocop
728 files inspected, no offenses detected

## Remote
`bundle exec rake test:remote TEST=test/remote/gateways/remote_checkout_v2_test.rb`
43 tests, 110 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### CODE REVIEW

**Comment by therufs** [comment]
Hey @AMHOL, 

This looks great so far! Could we, just for completeness, go ahead and pop the updated options into the corresponding remote test (`test_successful_purchase_with_additional_options`) too?

**Comment by AMHOL** [comment]
Thanks @therufs, I've amended the commit and force pushed to keep all changes in a single commit, see https://github.com/activemerchant/active_merchant/pull/4236/files#diff-997caba8ce528eaa5690041791c797b18f9727d588b5f26bee535dcf013f9469

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 1]
```diff
@@ -186,6 +188,17 @@ def add_3ds(post, options)
...
       end
 
+      def add_processing_channel(post, options)
+        post[:processing_channel_id] = options[:processing_channel_id] if options[:processing_channel_id]
+      end
+
+      def add_marketplace_data(post, options)
+        if options[:marketplace]
+          post[:marketplace] = {}
+          post[:marketplace][:sub_entity_id] = options[:marketplace][:sub_entity_id] if options[:marketplace][:sub_entity_id]
```
Comment (jessiagee) [inline lib/active_merchant/billing/gateways/checkout_v2.rb:198]: Let's check for nil with `options[:marketplace]` before attempting to dig into it

**Review by jessiagee — CHANGES_REQUESTED** [review]
Hi AMHOL! Let's fix the one area I pointed out than rebase and squash commits and I'll merge you into AM. Thank you!

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 1 above)
Comment (AMHOL) [reply lib/active_merchant/billing/gateways/checkout_v2.rb:198]: @jessiagee that is done on [line 196](https://github.com/activemerchant/active_merchant/pull/4236/files#diff-ac1e1ae60a6cc04111367e448e705b6e74104fd908e64d1ffe6c7237b5deb65dR196).

**Review by ajawadmirza — APPROVED** [review]
All tests passing. Good to go 👍

**Comment by AMHOL** [comment]
TY @jessiagee and @ajawadmirza

**Comment by jessiagee** [comment]
@AMHOL can you rebase this PR and squash the commits? Once that's done I'll approve and merge in. Thank you!

**Comment by AMHOL** [comment]
@jessiagee I've rebased and updated the description with the new local test run and rubocop data, but remote tests are failing on `test_partial_refund(RemoteCheckoutV2Test)` with a `403` error, I think it may be something to do with the secret key I'm using as I get the same failure on the master branch too.

**Comment by AMHOL** [comment]
@jessiagee remote tests are passing now:

43 tests, 110 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

**Review by jessiagee — APPROVED** [review]

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 1 above)
Comment (AMHOL) [reply lib/active_merchant/billing/gateways/checkout_v2.rb:203]: All good @jessiagee, thanks for the review.

**Pull request closed without merging**